File: _upstream/src/services/FileLogger.ts
import * as fs from "fs";
import * as path from "path";

export class FileLogger {
  private static logDir = path.join(process.cwd(), "logs");

  private static ensureDir() {
    if (!fs.existsSync(this.logDir)) {
      fs.mkdirSync(this.logDir, { recursive: true });
    }
  }

  private static getFilePath() {
    const date = new Date().toISOString().split("T")[0]; // YYYY-MM-DD
    return path.join(this.logDir, `bot-${date}.log`);
  }

  static log(level: "INFO" | "WARN" | "ERROR" | "TRADE", message: string) {
    this.ensureDir();
    const filePath = this.getFilePath();
    const timestamp = new Date().toISOString();
    // Strip ANSI color codes for clean text logs
    const cleanMessage = message.replace(/\u001b\[\d+m/g, "");
    const line = `[${timestamp}] [${level}] ${cleanMessage}\n`;

    fs.appendFile(filePath, line, (err) => {
      if (err) console.error("❌ Failed to write to log file:", err);
    });
  }
}

File: src/services/file_logger.go
package services

import (
	"fmt"
	"os"
	"path/filepath"
	"time"
)

// FileLogger 文件日志记录器
type FileLogger struct {
	logDir string
}

var fileLoggerInstance *FileLogger

func init() {
	fileLoggerInstance = &FileLogger{
		logDir: "logs",
	}
}

// ensureDir 确保日志目录存在
func (f *FileLogger) ensureDir() {
	if _, err := os.Stat(f.logDir); os.IsNotExist(err) {
		os.MkdirAll(f.logDir, 0755)
	}
}

// getFilePath 获取日志文件路径
func (f *FileLogger) getFilePath() string {
	f.ensureDir()
	date := time.Now().Format("2006-01-02")
	return filepath.Join(f.logDir, fmt.Sprintf("bot-%s.log", date))
}

// Log 记录日志
func (f *FileLogger) Log(level string, message string) {
	filePath := f.getFilePath()
	timestamp := time.Now().Format(time.RFC3339)
	line := fmt.Sprintf("[%s] [%s] %s\n", timestamp, level, message)

	file, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		fmt.Printf("❌ Failed to write to log file: %v\n", err)
		return
	}
	defer file.Close()

	file.WriteString(line)
}
